app/Repositories/Backend/Auth/RoleRepository.php

```diff
@@ -105,13 +105,17 @@ public function update(Role $role, array $data)
 
     /**
      * @param $name
+     * @param Role $exclude
      *
      * @return bool
      */
-    protected function roleExists($name)
+    protected function roleExists($name, $exclude = null)
     {
-        return $this->model
-                ->where('name', $name)
-                ->count() > 0;
+        $query = $this->model
+            ->where('name', $name);
+        if ($exclude !== null) {
+            $query->where('id', '!=', $exclude->id);
+        }
```

Comment:
Putting the $query in an else ($exclude === null) clause prevents it from running twice